engine/framework/registry/factory.go

```diff
@@ -84,13 +84,18 @@ func (f *SimpleWorkerFactory[T, C]) DeserializeConfig(configBytes []byte) (Worke
 	var config C
 	config = reflect.New(reflect.TypeOf(config).Elem()).Interface().(C)
 	if err := json.Unmarshal(configBytes, config); err != nil {
-		return nil, errors.Trace(fake.NewJobUnRetryableError(err))
+		return nil, errors.ErrDeserializeConfig.Wrap(err).GenWithStackByArgs()
 	}
 	return config, nil
 }
 
 // IsRetryableError implements WorkerFactory.IsRetryableError
 func (f *SimpleWorkerFactory[T, C]) IsRetryableError(err error) bool {
-	var errOut *fake.JobUnRetryableError
-	return !errors.As(err, &errOut)
+	if errors.Is(err, errors.ErrDeserializeConfig) {
+		return false
+	}
+	if strings.Contains(err.Error(), string(errors.ErrDeserializeConfig.RFCCode())) {
```

Comment:
when will this case be triggered?

Reply:
https://github.com/pingcap/tiflow/blob/e3cd4604d518f0474c7fe2d0f8fa9688dc18189f/engine/executor/server.go#L255
We may create an error from the error message and check whether it is retryable or not. However, I think it's not a good way to depend on the error message. This may be refactored in the future.